Core/Battler.gd

```diff
@@ -98,6 +98,6 @@ func _try_player_play_card_on_entity(entity: Entity) -> void:
 		if can_play:
 			# remove queued card, then play the card
 			# This is so the queued card doesn't have any influence over our hand count
```

Comment:
How does it affect the hand count then ? If we have 9 cards in hand, play draw 2 cards with a max of 10 cards in hand, how many cards are actually drawn ?
I would expect to draw 2 cards because to me the order should look like this:
- card is played, it's not in my hand
- my hand has 8 cards
- I draw 2 to a max of 10
- I discard the card I played (thus not drawing it)

Reply:
Ah you may be onto something here, if I play the Draw 2 at 9 cards in my hand, I only get one card.
https://github.com/Saplings-Projects/1M_sub/assets/5923102/0e2555a8-9a5a-461a-b2cf-03a65c98c49c

I'll look into this, but I think I'm going to have to split the queue vs. the discard out of the same function.

For other scenarios it does depend on number of cards in draw/discard.

For this scenario, we have no cards in discard or the draw so hence we do not draw anything.
https://github.com/Saplings-Projects/1M_sub/assets/5923102/d6e2d49e-5895-4e6e-98f7-6a3e986705e0

For this scenario, we play some cards that get discarded and playing the Draw 2 will shuffle those cards back into the draw pile and we draw again.
https://github.com/Saplings-Projects/1M_sub/assets/5923102/5fc6d96f-f142-40cd-ac9c-63190354018e

For this scenario, we have some cards already in the draw pile and play some cards in the discard. Draw 2 will just pull from the Draw pile.
https://github.com/Saplings-Projects/1M_sub/assets/5923102/2bb21aa2-6ced-4fc1-afec-166f14fe9d1a